### PR TITLE
Fix for the the fallback issue

### PR DESCRIPTION
**Pull request by Sfurti-yb**
These code changes resolve the following issue that were reported related to fallback feature:

If all nodes in PRIMARY_PLACEMENT are down, connection go to FIRST_FALLBACK nodes, now if a node in PRIMARY_PLACEMENT come back up, all new connections do not go to that node but are rather equally distributed between this node and FIRST_FALLBACK nodes.

### CODE REVIEW

File: tests/test_fallback_topology.py [hunk 1]
```diff
@@ -1,35 +1,40 @@
...
+    yb_path = '/Users/ssarah/code/yugabyte-db'
     os.system(yb_path+'/bin/yb-ctl destroy')
     os.system(yb_path+'/bin/yb-ctl create --rf 3 --placement_info \"aws.us-west.us-west-2a,aws.us-west.us-west-2b,aws.us-west.us-west-2c\"')
 
-def createClusterWithSixNodes():
+def create_cluster_with_six_nodes():
     global yb_path
-    yb_path = os.getenv('YB_PATH')
+    # yb_path = os.getenv('YB_PATH')
+    yb_path = '/Users/ssarah/code/yugabyte-2.18.0.0'
```
Comment (ashetkar) [inline tests/test_fallback_topology.py:22]: Why not read from the environment variable?

File: lib/policies.py [hunk 2]
```diff
@@ -68,6 +73,21 @@ def getLeastLoadedServer(self,failedhosts):
         
         if chosenHost != '':
             self.updateConnectionMap(chosenHost,1)
+        elif self.useHostColumn == None:
+            newList = []
+            newList = newList + self.currentPublicIps
+            if not newList:
```
Comment (ashetkar) [inline lib/policies.py:79]: Should this be `if newList:` ?

File: lib/policies.py [hunk 3]
```diff
@@ -118,14 +145,31 @@ def refresh(self, conn):
         if not self.needsRefresh():
             return True
         currTime = time.time()
+        self.lastServerListFetchTime = currTime
+        possiblyReachableHosts = []
+        for host, time_inactive in self.unreachableHosts.items():
+            if (currTime - time_inactive) > self.DEFAULT_FAILED_HOST_TTL_SECONDS:
```
Comment (ashetkar) [inline lib/policies.py:151]: Is this configurable?

Comment (Sfurti-yb) [reply lib/policies.py]: Made it configurable.

**Review by ashetkar — APPROVED** [review]
LGTM with a minor comment and formatting.

**Pull request merged**